src/pint/pintk/pulsar.py

```diff
@@ -86,6 +86,9 @@ def __init__(self, parfile=None, timfile=None, ephem=None):
 
         # turns pre-existing jump flags in toas.table['flags'] into parameters in parfile
         self.prefit_model.jump_flags_to_params(self.all_toas)
+        # adds flags to toas.table for existing jump parameters from .par file
+        if "PhaseJump" in self.prefit_model.components:
```

Comment:
It doesn't remove components, the if statement is there in case the loaded .par file does not have any jumps. If there is no jump component, there is no object that defines the function within the timing model so an error will be thrown if the function is called on the timing model.

Reply:
I'm just checking what happens - because the component is not removed, this branch will be run every time we get here. I think the operation is idempotent so I think it should be fine?

Reply:
But if there is no PhaseJump object to begin with, running jump_params_to_flags will throw an error because there is no object initialized which defines it.